src/map/pc.c

```diff
@@ -9678,44 +9693,68 @@ bool pc_unequipitem(struct map_session_data *sd, int n, int flag) {
 	int i, iflag;
 	bool status_cacl = false;
 
-	nullpo_retr(false,sd);
+	nullpo_retr(false, sd);
 
 	if (n < 0 || n >= MAX_INVENTORY) {
-		clif_unequipitemack(sd,0,0,0);
+		clif_unequipitemack(sd, 0, 0, 0);
 		return false;
 	}
 	if (!sd->status.inventory[n].equip) {
-		clif_unequipitemack(sd,n,0,0);
+		clif_unequipitemack(sd, n, 0, 0);
 		return false; //Nothing to unequip
 	}
 	// status change that makes player cannot unequip equipment
-	if (!(flag&2) && sd->sc.count &&
+	if (!(flag & 2) && sd->sc.count &&
 		(sd->sc.data[SC_BERSERK] ||
 		sd->sc.data[SC_SATURDAYNIGHTFEVER] ||
 		sd->sc.data[SC__BLOODYLUST] ||
 		sd->sc.data[SC_KYOUGAKU] ||
 		(sd->sc.data[SC_PYROCLASTIC] &&
 		sd->inventory_data[n]->type == IT_WEAPON)))	// can't switch weapon
 	{
-		clif_unequipitemack(sd,n,0,0);
+		clif_unequipitemack(sd, n, 0, 0);
 		return false;
 	}
 
 	if (battle_config.battle_log)
-		ShowInfo("unequip %d %x:%x\n",n,pc_equippoint(sd,n),sd->status.inventory[n].equip);
+		ShowInfo("unequip %d %x:%x\n", n, pc_equippoint(sd, n), sd->status.inventory[n].equip);
 
-	for(i = 0; i < EQI_MAX; i++) {
+	for (i = 0; i < EQI_MAX; i++) {
 		if (sd->status.inventory[n].equip & equip_bitmask[i])
 			sd->equip_index[i] = -1;
 	}
 
-	if(sd->status.inventory[n].equip & EQP_HAND_R) {
+	if (sd->status.inventory[n].equip & EQP_HAND_R) {
 		sd->weapontype1 = 0;
 		sd->status.weapon = sd->weapontype2;
 		pc_calcweapontype(sd);
-		clif_changelook(&sd->bl,LOOK_WEAPON,sd->status.weapon);
-		if( !battle_config.dancing_weaponswitch_fix )
+		clif_changelook(&sd->bl, LOOK_WEAPON, sd->status.weapon);
+		if (!battle_config.dancing_weaponswitch_fix)
 			status_change_end(&sd->bl, SC_DANCING, INVALID_TIMER); // Unequipping => stop dancing.
+
+		// [benching] 
+		switch (battle_config.weapon_ammo_unequip){
```

Comment:
the switch here is not really extendable with a bitmask.
I mean the code is correct don't get me wrong. 
But I want to add an option to that weapon_ammo_unequip, I may need to add 3 more case to fill all the combinaison, and so on...
Imo I'd just have writting this with 2 independant if.